code/game/objects/items/mail.dm

```diff
@@ -0,0 +1,305 @@
+		var/list/job_goodies = this_job.get_mail_goodies()
+		if(LAZYLEN(job_goodies))
+			// certain roles and jobs (prisoner) do not receive generic gifts.
+			if(this_job.exclusive_mail_goodies)
+				goodies = job_goodies
+			else
+				goodies += job_goodies
+
+	for(var/iterator in 1 to goodie_count)
+		var/target_good = pickweight(goodies)
+		var/atom/movable/target_atom = new target_good(src)
+		body.log_message("[key_name(body)] received [target_atom.name] in the mail ([target_good])", LOG_GAME)
+
+	return TRUE
+
+/// Alternate setup, just complete garbage inside and anyone can open
+/obj/item/mail/proc/junk_mail()
+
+	var/obj/junk = /obj/item/paper/fluff/junkmail_generic
+	var/special_name = FALSE
+
+	if(prob(25))
+		special_name = TRUE
+		junk = pick(list(/obj/item/paper/pamphlet/gateway, /obj/item/paper/pamphlet/centcom/visitor_info, /obj/item/paper/fluff/junkmail_redpill, /obj/effect/decal/cleanable/ash))
+
+	var/list/junk_names = list(
+		/obj/item/paper/pamphlet/gateway = "[initial(name)] for [pick(GLOB.adjectives)] adventurers",
+		/obj/item/paper/pamphlet/centcom/visitor_info = "[initial(name)] for info about visiting nanotrasen space stations",
+		/obj/item/paper/fluff/junkmail_redpill = "[initial(name)] for those feeling [pick(GLOB.adjectives)] working at Nanotrasen",
+		/obj/effect/decal/cleanable/ash = "[initial(name)] with INCREDIBLY IMPORTANT ARTIFACT- DELIVER TO SCIENCE DIVISION. HANDLE WITH CARE.",
+	)
+
+	color = pick(department_colors) //eh, who gives a shit.
+	name = special_name ? junk_names[junk] : "important [initial(name)]"
+
+	junk = new junk(src)
+	return TRUE
+
+/obj/item/mail/proc/disposal_handling(disposal_source, obj/structure/disposalholder/disposal_holder, obj/machinery/disposal/disposal_machine, hasmob)
+	if(!hasmob)
+		disposal_holder.destinationTag = sort_tag
+
+/// Subtype that's always junkmail
+/obj/item/mail/junkmail/Initialize(mapload)
+	. = ..()
+	junk_mail()
+
+/// Crate for mail from CentCom.
+/obj/structure/closet/crate/mail
+	name = "mail crate"
+	desc = "A certified post crate from CentCom."
+	icon_state = "mail"
+
+/obj/structure/closet/crate/mail/update_icon()
+	. = ..()
+	if(opened)
+		icon_state = "[initial(icon_state)]open"
+		if(locate(/obj/item/mail) in src)
+			icon_state = initial(icon_state)
+	else
+		icon_state = "[initial(icon_state)]sealed"
+
+/// Fills this mail crate with N pieces of mail, where N is the lower of the amount var passed, and the maximum capacity of this crate. If N is larger than the number of alive human players, the excess will be junkmail.
+/obj/structure/closet/crate/mail/proc/populate(amount)
+	var/mail_count = min(amount, storage_capacity)
+	// Fills the
+	var/list/mail_recipients = list()
+
+	for(var/mob/living/carbon/human in GLOB.player_list)
+		if(human.stat == DEAD || !human.mind)
+			continue
+		// Skip wizards, nuke ops, cyborgs; Centcom does not send them mail
+		if(human.faction.Find("Station", 1, 0) != 0)
```

Comment:
Not sure about wizards and nuke ops, but silicons don't have the Station faction, and therefore can be sent mail